src/Comparator/BackwardsCompatibility/MethodBased/MethodScopeChanged.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\ApiCompare\Comparator\BackwardsCompatibility\MethodBased;
+
+use Roave\ApiCompare\Change;
+use Roave\ApiCompare\Changes;
+use Roave\BetterReflection\Reflection\ReflectionMethod;
+use function sprintf;
+
+/**
+ * A method that changes from instance to static or the opposite has to be called differently,
+ * so any of such changes are to be considered BC breaks
+ */
+final class MethodScopeChanged implements MethodBased
+{
+    public function compare(ReflectionMethod $fromMethod, ReflectionMethod $toMethod) : Changes
+    {
+        if ($fromMethod->isPrivate()) {
```

Comment:
Didn't this already get extracted in `AccessibleMethodFunctionBasedChange`?

Reply:
Same as https://github.com/Roave/ApiCompare/pull/38#discussion_r182383370 - covariance/contravariance are concepts that need to be known/learned: no escape from that.

Reply:
Interestingly, this is extremely useful when I'm going to introduce a `TraitBased` comparator, because the same checks can now be used both for checking `private` and `protected`/`public` changes.

I made an `AccessiblePropertyChanged` for that for now :+1: